java-frontend/src/main/java/org/sonar/java/model/JWarning.java

```diff
@@ -89,7 +90,7 @@ boolean isMatching(IProblem warning) {
             warning.getSourceLineNumber(),
             astNode.getColumnNumber(warning.getSourceStart()),
             astNode.getLineNumber(warning.getSourceEnd()),
-            astNode.getColumnNumber(warning.getSourceEnd()));
+            astNode.getColumnNumber(warning.getSourceEnd()) + 1);
```

Comment:
was also noticeable in review on PEACH. The end of highlighting was wrong by 1 character.